### PR TITLE
Remove circular dependency with isAllowedResource function

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

### CODE REVIEW

File: src/common/utils/allowed-resource.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  const resources = [resource].flat();
+  const cluster = ClusterStore.getInstance().getById(getHostedClusterId());
+
+  if (!cluster?.allowedResources) {
+    return false;
+  }
+
+  if (resources.length === 0) {
+    return true;
+  }
```
Comment (jim-docker) [inline src/common/utils/allowed-resource.ts:36]: guess you could do this before getting the cluster

File: src/common/utils/allowed-resource.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  const allowedResources = new Set(cluster.allowedResources);
+
+  return resources.every(resource => allowedResources.has(resource));
```
Comment (jim-docker) [inline src/common/utils/allowed-resource.ts:40]: Why is this better than:
```suggestion
  return resources.every(resource => cluster.allowedResources.includes(resource));
```

**Review by jim-docker — APPROVED** [review]

File: src/common/utils/allowed-resource.ts (hunk 1 above)
Comment (Nokel81) [reply src/common/utils/allowed-resource.ts:36]: Fair enough

File: src/common/utils/allowed-resource.ts (hunk 2 above)
Comment (Nokel81) [reply src/common/utils/allowed-resource.ts:40]: Generally because I would figure it is less work (though I don't have the data to back it up).

**Comment by Nokel81** [comment]
Will make a follow up PR once everything is merged.

**Pull request merged**